### DISCUSSION TITLE
What's the intended string format for grammars for endpoints?

### DISCUSSION THREAD

**Discussion by Calandiel** [LostRuins/koboldcpp#533] [Q&A]
Hello! ^^
I've been playing with the api endpoints (in `particular api/v1/generate`) and I noticed that no matter what I do, it seems that koboldcpp inserts escapes before backslashes, even if the string sent already does all the escaping.

To illustrate, my payload is:
```
{"prompt": "Test ", "grammar": "root    ::= (\"true\" | \"false\" | \"null\")
"}
```
However, when I run koboldcpp, it claims that the grammar was:
```
"grammar": "root    ::= (\\"true\\" | \\"false\\" | \\"null\\")\n"
```
That exact same grammar, when supplied through the webui looks like this:
```
"grammar": "root    ::= (\"true\" | \"false\" | \"null\")\n"
```
Note the lack of double escapes.
I'm not sure how stable grammar support is but I did my best to verify that what's posted to the api endpoint has single escapes. I was going to open an issue about it but I thought I'd ask here first (as I imagine I could simply be overlooking something). Am I doing something obviously wrong? Or maybe that's just not what grammars expect as inputs?

**Comment by LostRuins** [comment]
Grammar has worked correctly so far for me. Here's an example GBNF file: https://github.com/ggerganov/llama.cpp/blob/master/grammars/chess.gbnf 

try pasting that into the webui, then generating?

**Reply by LostRuins** [reply]
I am not handling quotes in any special way when formatting the JSON file. I simply call the serialize function, and allow it to escape the string automatically.

**Reply by Calandiel** [reply]
Thanks for replying ^^ Just to clarify, the grammar works fine in the webui. I only notice issues when accessing the api endpoint directly
